File: web/sites/domain/vas/agreement.php
<?php
	class Agreement
	{
		public	$id,
				$partner_id,
				$name,
				$date_created,
				$finder_fee,
				$revenue_share,
				$product_sms,
				$product_voice,
				$product_games,
				$product_vg,
				$product_others,
				$start_date,
				$end_date;

		public function __construct($data)
		{
			$this->id = get_value_from_array('ID', $data);
			$this->partner_id = get_value_from_array('PartnerID', $data);
			$this->name = get_value_from_array('Name', $data);
			$this->finder_fee = get_value_from_array('FinderFee', $data);
			$this->revenue_share = get_value_from_array('RevenueShare', $data);
			$this->product_sms = get_value_from_array('ProductSMS', $data);
			$this->product_voice = get_value_from_array('ProductVoice', $data);
			$this->product_games = get_value_from_array('ProductGames', $data);
			$this->product_vg = get_value_from_array('ProductVG', $data);
			$this->product_others = get_value_from_array('ProductOthers', $data);
			$this->start_date = get_value_from_array('StartDate', $data);
			$this->end_date = get_value_from_array('EndDate', $data);
			$this->date_created = get_value_from_array('DateCreated', $data);
		}

		public function start_date()
		{
			return date("d M Y", strtotime($this->start_date));
		}

		public function end_date()
		{
			return date("d M Y", strtotime($this->end_date));
		}
	}
?>

File: web/ror/app/models/agreement.rb
class Agreement

    @id
    @partner
    @name
    @dateCreated
    @finderFee
    @revenueShare
    @startDate
    @endDate
    
    attr_accessor :id, :name, :dateCreated, :finderFee, :revenueShare, :startDate, :endDate
    
    def initialize(args={})
        @id = args[:id] || nil
        @name = args[:name] || nil
        @partner = args[:partner] || nil

        @finderFee = args[:finderFee] || nil
        @revenueShare = args[:revenueShare] || nil
        
        @startDate = args[:startDate] || nil
        @endDate = args[:endDate] || nil

        begin
            @dateCreated = DateTime.strptime(args[:dateCreated], "%m/%d/%Y").to_time unless args[:dateCreated].nil? 
        rescue ArgumentError => e
            @dateCreated = nil
        end
    end
    
    
    def self.create(args={})
        #
        # Create agreement
        #
    
        db = Dbconn.get_instance('master')
        
        stmt = db.prepare "INSERT INTO partneragreement 
                                       (Name, PartnerId, FinderFee, RevenueShare, StartDate, EndDate, DateCreated) 
                                VALUES (?, ?, ?, ?, ?, ?, NOW())"
        
        stmt.execute args[:name], args[:partnerId], args[:finderFee], args[:revenueShare], args[:startDate], args[:endDate]
                     
        @agreementId = stmt.insert_id
        
        stmt.close
    
        return @agreementId
    end
    
    
    def self.find_by_partner_id(partner_id)
        #
        # Find agreements associated with a partner
        #
        
        db = Dbconn.get_instance('master')
        
        stmt = db.prepare "SELECT pa.Id, pa.Name, date_format(pa.DateCreated, '%m/%d/%Y') as AgreementDateCreated,
                                  pa.FinderFee, pa.RevenueShare, 
                                  date_format(pa.StartDate,'%m/%d/%Y') as AgreementStartDate, 
                                  date_format(pa.EndDate,'%m/%d/%Y') as AgreementEndDate
                             FROM partneragreement pa
                            WHERE pa.PartnerId = ?
                         ORDER BY pa.StartDate DESC"
        stmt.execute partner_id
        
        agreements = []
        
        while row = stmt.fetch do
        
            agreement = Agreement.new(
                            :id => row[0],
                            :name => row[1],
                            :dateCreated => row[2],
                            :finderFee => row[3],
                            :revenueShare => row[4],
                            :startDate => row[5],
                            :endDate => row[6]
                        )
                
            agreements << agreement
        end 
        
        stmt.free_result
        stmt.close  
        
        return agreements
    end
    
    
    def self.find(agreement_id)
        #
        # Find agreement
        #
        
        db = Dbconn.get_instance('master')
        stmt = db.prepare "SELECT pa.Id, pa.Name, date_format(pa.DateCreated, '%m/%d/%Y') as AgreementDateCreated,
                                  pa.FinderFee, pa.RevenueShare, 
                                  date_format(pa.StartDate,'%m/%d/%Y') as AgreementStartDate, 
                                  date_format(pa.EndDate,'%m/%d/%Y') as AgreementEndDate
                             FROM partneragreement pa
                            WHERE pa.Id = ?"
        stmt.execute agreement_id
        
        agreement = nil
        if row = stmt.fetch 
            agreement = Agreement.new(
                            :id => row[0],
                            :name => row[1],
                            :dateCreated => row[2],
                            :finderFee => row[3],
                            :revenueShare => row[4],
                            :startDate => row[5],
                            :endDate => row[6]
                        )
        end
        
        stmt.free_result
        stmt.close
        
        return agreement
    end
    
    def self.add_build(agreement_id, build_id)
        #
        # Add build to the agreement
        #
        
        db = Dbconn.get_instance('master')
        stmt = db.prepare "INSERT INTO partneragreementbuild (PartnerAgreementID, PartnerBuildID) 
                            SELECT ?, ? FROM DUAL  
                            WHERE NOT EXISTS 
                            (SELECT * FROM partneragreementbuild WHERE PartnerAgreementID = ? AND PartnerBuildID = ?)"
        stmt.execute agreement_id, build_id, agreement_id, build_id
        
        stmt.close
    end
    
    def self.remove_build(agreement_id, build_id)
        #
        # Remove build in the agreement
        #
        
        db = Dbconn.get_instance('master')
        stmt = db.prepare "DELETE FROM partneragreementbuild WHERE PartnerAgreementID=? AND PartnerBuildID=?"
        stmt.execute agreement_id, build_id
        
        stmt.close
    end
end